src/System.Management.Automation/commands/utility/FormatAndOutput/out-console/OutConsole.cs

```diff
@@ -79,9 +79,14 @@ protected override void BeginProcessing()
                 mrt.MergeUnclaimedPreviousErrorResults = true;
             }
 
-            _savedTranscribeOnly = Host.UI.TranscribeOnly;
+            if (!_transcriptStateChanged)
+            {
+                _savedTranscribeOnly = Host.UI.TranscribeOnly;
+            }
+
             if (Transcript)
             {
+                _transcriptStateChanged = true;
```

Comment:
You never set `_transcriptStateChanged` back to `false`. So `Out-Default` stop capturing `TranscribeOnly` after first call with `-Transcript` switch.